pkg/operator/ceph/object/user.go

```diff
@@ -162,6 +170,26 @@ func CreateUser(c *Context, user ObjectUser, force bool) (*ObjectUser, int, erro
 	return decodeUser(result)
 }
 
+// CreateOrRecreateUserIfExists deletes an existing user if it already exists and re-creates it
+func CreateOrRecreateUserIfExists(c *Context, user ObjectUser, force bool) (*ObjectUser, int, error) {
```

Comment:
Why do we have do delete and recreate a user? Why not just create it if it doesn't exist?

Reply:
It was easier to accomplish with the existing code, otherwise we need to manage the access key(s) of the dashboard-admin user. I haven't found code to manage the radosgw users access/secret key(s) in Rook (yet?).

Reply:
From the caller of this method, it looks like we only get to this method if the dashboard hasn't already been configured. If the dashboard was already configured, this method is not called again. So why would the dashboard user need to be re-created? It doesn't seem like we would hit this code in that case. But I must still be missing the scenario. Could you add comments to the code to explain?

Reply:
`CreateOrRecreateUserIfExists()` will be called under the following conditions:
* `dashboard-admin` user does not exist.
* `dashboard-admin` user exists but the credentials that the dashboard has set and the user has assigned are different.
	* For this case can either manage the credentials or as I wrote delete and re-create the user with the correct credentials.

The idea here is that we ensure the dashboard-admin user is always correctly configured, even when the credentials are changed for whatever reason. I chose to re-create the user instead of managing the access/secret keys for it, was out of simplicity as we more or less have the code for creating/deleting users already and deleting.

Reply:
Where is the check for the changed creds? I just saw the check for the existence of the user. thanks

Reply:
@travisn Here https://github.com/rook/rook/pull/10137/files#diff-3d31541d76db0087520217d3ccb2b5d704b3a7e05a31167c80bcf69b2cfcaec8R821-R840
The dashboard's credentials are pulled in from here: https://github.com/rook/rook/pull/10137/files#diff-3d31541d76db0087520217d3ccb2b5d704b3a7e05a31167c80bcf69b2cfcaec8R843-R869

Reply:
Thanks for the references, I see now. Perhaps this func name would just help.
```suggestion
func CreateOrRecreateUserIfCredsChanged(c *Context, user ObjectUser, force bool) (*ObjectUser, int, error) {
```

Reply:
Good point. How about adding that detail to the comments where this is called? I don't see comments in the calling method about only doing it if creds changed for an existing user.

Reply:
@travisn I have added it to the comment of the `CreateOrRecreateUserIfExists()` function